automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookPage.scala

```diff
@@ -174,49 +175,51 @@ class NotebookPage(override val url: String)(override implicit val authToken: Au
     await condition isKernelShutdown
   }
 
+  /**
+    * Throw TimeoutException if Kernel is not ready after restart when timeout is reached
+    *
+    * @param timeout
+    */
   def restartKernel(timeout: FiniteDuration = 1 minute): Unit = {
     click on kernelMenu
     click on (await enabled restartKernelSelection)
     click on (await enabled restartKernelConfirmationSelection)
-    waitForIdleKernel(timeout)
-  }
-
-  /**
-    * wait for kernel icon to become idle. default timeout == 1.minute
-    */
-  def waitForIdleKernel(timeout: FiniteDuration = 1.minutes): Unit = {
-    await condition (
-      !cellsAreRunning && isKernelIdle() &&
-        find(id("notification_kernel")).exists(_.underlying.getCssValue("display") == "none"), timeout.toSeconds
-    )
-    await condition (find(id("kernel_indicator_icon")).exists(_.isDisplayed), 2) // little time to check isDisplayed to prevent StaleWebElementException
+    await notVisible restartKernelConfirmationSelection
+    await condition (isKernelReady && kernelNotificationText() == "none", timeout.toSeconds)
   }
 
   /**
     * wait for kernel to become ready. default timeout == 1.minute
+    *
+    * Throw TimeoutException if Kernel is not ready when timeout is reached
     */
-  def awaitReadyKernel(timeout: FiniteDuration = 1.minute): Unit = {
-    try {
-      waitForIdleKernel(timeout)
-    } catch {
-      case _: TimeoutException =>
-        if (isKernelDisconnected()) {
-          restartKernel(timeout)
-        }
+  def waitReadyKernel(timeout: FiniteDuration = 1.minutes): Unit = {
+    Try {
+      await condition (
+        !cellsAreRunning &&
+          isKernelReady() &&
+          kernelNotificationText() == "none", timeout.toSeconds
+      )
+    } recover {
+      case e: TimeoutException => restartKernel(timeout)
     }
   }
 
   def clickRunCell(timeout: FiniteDuration = 2.minutes): Unit = {
     click on runCellButton
-    awaitReadyKernel(timeout)
+    waitReadyKernel(timeout)
   }
 
   def isKernelDisconnected(): Boolean = {
     find(id("kernel_indicator_icon")).exists(_.underlying.getAttribute("class") == "kernel_disconnected_icon")
   }
 
-  def isKernelIdle(): Boolean = {
+  def isKernelReady(): Boolean = {
     find(id("kernel_indicator_icon")).exists(_.underlying.getAttribute("class") == "kernel_idle_icon")
   }
 
+  def kernelNotificationText(): String = {
```

Comment:
Scala convention for `def`s that don't take parameters. Use `()` if the method performs a side-effect and don't use `()` otherwise. If `find` below doesn't do something side-effecty (like opening a new browser, etc.), there shouldn't be parentheses.

Same for `isKernelReady`.